### PR TITLE
KOGITO-994: Load more for Domain Explorer

### PR DESCRIPTION
**Pull request by nmirasch**
https://issues.redhat.com/browse/KOGITO-994

- Add load more to domain explorer view
- Encapsulate design and 'loading' state on LoadMoreComponent
- Updated DataListContainer use of LoadMoreComponent

### CODE REVIEW

File: packages/management-console/src/components/Atoms/LoadMoreComponent/LoadMoreComponent.css [hunk 1]
```diff
@@ -0,0 +1,4 @@
+.kogito-management-console-load-more{
```
Comment (cristianonicolai) [inline packages/management-console/src/components/Atoms/LoadMoreComponent/LoadMoreComponent.css]:
```suggestion
.kogito-management-console-load-more {
```

File: packages/management-console/src/components/Atoms/LoadMoreComponent/LoadMoreComponent.tsx [hunk 2]
```diff
@@ -35,18 +44,32 @@ const LoadMoreComponent: React.FC<IOwnProps> = ({
 
   return (
     <React.Fragment>
```
Comment (cristianonicolai) [inline packages/management-console/src/components/Atoms/LoadMoreComponent/LoadMoreComponent.tsx]: @nmirasch do we still need to wrap it inside a _React.Fragment_ ?

File: packages/management-console/src/components/Templates/DomainExplorerDashboard/DomainExplorerDashboard.tsx [hunk 3]
```diff
@@ -35,6 +36,7 @@ const DomainExplorerDashboard = props => {
     (props.location.state && props.location.state.parameters) || [];
   const rememberedSelections =
     (props.location.state && props.location.state.selected) || [];
+  const pSize = 10;
```
Comment (cristianonicolai) [inline packages/management-console/src/components/Templates/DomainExplorerDashboard/DomainExplorerDashboard.tsx]: I might be wrong but I guess the idea is to use pageSize useState instead?

File: packages/management-console/src/components/Atoms/LoadMoreComponent/LoadMoreComponent.tsx [hunk 4]
```diff
@@ -1,18 +1,27 @@
...
+  DataListItem,
+  DataListCell,
+  Spinner
+} from '@patternfly/react-core';
+import './LoadMoreComponent.css';
 
 interface IOwnProps {
   offset: number;
   setOffset: (offset: number) => void;
   getProcessInstances: (initval: number, pageSize: number) => void;
```
Comment (cristianonicolai) [inline packages/management-console/src/components/Atoms/LoadMoreComponent/LoadMoreComponent.tsx]: maybe we should call this like callback?

File: packages/management-console/src/components/Organisms/DomainExplorerColumnPicker/DomainExplorerColumnPicker.tsx [hunk 5]
```diff
@@ -355,9 +382,18 @@ const DomainExplorerColumnPicker: React.FC<IOwnProps> = ({
     return unique;
   }
 
+  const onResetQuery = _parameters => {
+    setOffsetVal(0);
+    offsetVal = 0;
```
Comment (cristianonicolai) [inline packages/management-console/src/components/Organisms/DomainExplorerColumnPicker/DomainExplorerColumnPicker.tsx:387]: similar here, I think we need to use the useState hooks only, no?

File: packages/management-console/src/components/Atoms/LoadMoreComponent/LoadMoreComponent.css (hunk 1 above)
Comment (nmirasch) [reply packages/management-console/src/components/Atoms/LoadMoreComponent/LoadMoreComponent.css]: done!

File: packages/management-console/src/components/Atoms/LoadMoreComponent/LoadMoreComponent.tsx (hunk 2 above)
Comment (nmirasch) [reply packages/management-console/src/components/Atoms/LoadMoreComponent/LoadMoreComponent.tsx]: removed, yes it wasn't necessary.  :-)

File: packages/management-console/src/components/Templates/DomainExplorerDashboard/DomainExplorerDashboard.tsx (hunk 3 above)
Comment (nmirasch) [reply packages/management-console/src/components/Templates/DomainExplorerDashboard/DomainExplorerDashboard.tsx]: I have updated this renaming to defaultPageSize and adding useState

**Comment by cristianonicolai** [comment]
@nmirasch maybe add some mocked data to see it in action? :)

**Comment by srambach** [comment]
I'm seeing "Load 10 more" not actually add 10 more, but instead it reloads just 10 without the loading buttons. 

The "Load 20 more" button does add more but I can't tell if it would add 20 more for a total of 30 since there aren't that many. 

Also, there is quite a delay in loading. Can we add the spinner to the button while loading, then redraw the screen? The -xs size spinner should fit into the button. https://www.patternfly.org/v4/design-guidelines/usage-and-behavior/data-loading#main-content Or, we could use a spinner below the list, indicating more are being fetched, but from a UX standpoint, it's probably easier to show on the button.

**Comment by nmirasch** [comment]
@cristianonicolai  I'm trying to add more mock data to see this in action but for the moment the testing have to be against the remote data index with real data.. .
@srambach  
- About the message 'Load X more' are the same as was in Process instances explorer, in fact it's using the same component. But you mean we have to change the 'Load X more" -> "Add X more"?
- the 'load more' buttons bar would appear if could be more that the showed ones.. The point is not to calculate how much items are there. Those buttons meaning would be something like 'Try to load 20 more".
- I don't understand the last thing about the spinner. It have to be shown when the query to load more items is performed, let me re-check that..

**Comment by srambach** [comment]
Sorry :)  what I'm trying to say is:
- When the "Load X more" button is pressed, I'd expect to get X _more_ than I see now, unless that is more than there are in total.
- When I click the button, first the buttons go away, then after a pause, I see more items in the list. What would be better is immediately after the button is clicked, a spinner is added to the button to show work in progress, and then the new items are loaded. If there are no more items, then the buttons can go away when the new items are added to the list. Does that make sense? Is it possible?

**Comment by nmirasch** [comment]
@srambach thanks for your comments, makes absolutely sense to me . We can go throw this at today meeting, to see this in action and see what have to be different, maybe there are some bug that I'm missing right now.

**Review by cristianonicolai — APPROVED** [review]

File: packages/management-console/src/components/Atoms/LoadMoreComponent/LoadMoreComponent.css [hunk 6]
```diff
@@ -0,0 +1,4 @@
+.kogito-management-console-load-more {
```
Comment (srambach) [inline packages/management-console/src/components/Atoms/LoadMoreComponent/LoadMoreComponent.css]: We haven't been 100% consistent with class names, but let's try to stick with `kogito-management-console__` plus the component (and any elements within that are needed) for now, so something like `kogito-management-console__load-more` here. (and where it's used)

File: packages/management-console/src/components/Atoms/LoadMoreComponent/LoadMoreComponent.css [hunk 7]
```diff
@@ -0,0 +1,4 @@
...
+  margin-bottom: 24px;
```
Comment (srambach) [inline packages/management-console/src/components/Atoms/LoadMoreComponent/LoadMoreComponent.css]: We can use our spacer variables here
`margin-bottom: var(--pf-global--spacer--lg);`

**Review by AjayJagan — APPROVED** [review]

File: packages/management-console/src/components/Atoms/LoadMoreComponent/LoadMoreComponent.css (hunk 7 above)
Comment (nmirasch) [reply packages/management-console/src/components/Atoms/LoadMoreComponent/LoadMoreComponent.css]: done

File: packages/management-console/src/components/Atoms/LoadMoreComponent/LoadMoreComponent.css (hunk 6 above)
Comment (nmirasch) [reply packages/management-console/src/components/Atoms/LoadMoreComponent/LoadMoreComponent.css]: updated!

**Comment by nmirasch** [comment]
@srambach  Thanks, css updated! @AjayJagan I also added some LoadMoreComponet tests too.

**Review by srambach — APPROVED** [review]

**Comment by Sara4994** [comment]
@nmirasch looks good to me :+1:

**Review by Sara4994 — APPROVED** [review]

**Pull request merged**